### ISSUE TITLE
Is there any way to get this working with Flutter Web?

### ISSUE THREAD

**Issue by agdsat134** [InvisibleWrench/FlutterMidiCommand#17]
If it isn't supported, are there plans to support it in the future?

**Comment by mortenboye** [comment]
I guess it could potentially be supported. At least on the browsers that supports the Web MIDI API. No plans at this point though, but happy to consider it if there is a need.

**Comment by systemcode** [comment]
I'm using FlutterMidiCommand for an iOS/Androd app I am working on. If you get this to work on Web that would be great for me too. (also MacOS and Win?)
BTW - thanks for your work on this package.

**Comment by narrsam** [comment]
I would also be interested

**Comment by davebound** [comment]
+1

**Comment by mortenboye** [comment]
Currently blocked by https://github.com/dart-lang/sdk/issues/33248

the bridge to html does not initialise input and output maps correctly, but return empty maps.

**Comment by mortenboye** [comment]
> I'm using FlutterMidiCommand for an iOS/Androd app I am working on. If you get this to work on Web that would be great for me too. (also MacOS and Win?)
> BTW - thanks for your work on this package.

macOS support has just been released.

**Comment by ifredom** [comment]
I render with WebView, and flutter passes information to WebView

**Comment by systemcode** [comment]
> 
> macOS support has just been released.

Does not seem to work in 0.2.4

[!] CocoaPods could not find compatible versions for pod "flutter_midi_command":
  In Podfile:
    flutter_midi_command (from `Flutter/ephemeral/.symlinks/plugins/flutter_midi_command/macos`)

Specs satisfying the `flutter_midi_command (from `Flutter/ephemeral/.symlinks/plugins/flutter_midi_command/macos`)` dependency were found, but they required a higher minimum deployment target.

**Comment by mortenboye** [comment]
@systemcode what is your deployment target?

**Comment by systemcode** [comment]
> @systemcode what is your deployment target?
> 
> 
flutter run -d macos

Looks like everything is set up correctly to me.

**Comment by mortenboye** [comment]
@systemcode 
The example project works for me on 10.15.7

Please file a separate issue regarding macos.

**Comment by stuartkuentzel** [comment]
Any update on this since the Flutter Web stable release?

**Comment by mortenboye** [comment]
@stuartkuentzel No, it seems nothing relevant have changed in the html API.
I think package:js/js.dart could be a way to work around this, but haven't had a chance to dive into it. 

Contributions are welcome

**Comment by maks** [comment]
Just an update on this, I'm using a variation on [the work around from this comment](https://github.com/dart-lang/sdk/issues/33248#issuecomment-914223643) successfully with a flutter web app at the moment.

My "variation" is that I've got the "wrapper" code in a JS file versus doing it purely from Dart code using dart:js but that should be just an implementation detail.

My code is pretty rough at the moment (eg. only deals with 1 midi device) but I can try to clean it up and submit a PR if you would be interested in including it @mortenboye ?

As a side note, it also seems to work ok with the brand new support for WebMidi that Firefox has just added in v97 (currently in nightly builds only).

**Comment by maks** [comment]
Sorry forgot to include a link to my [current workaround code](https://github.com/maks/e2_edit/blob/main/web/js/midi.js) in [the Flutter app project ](https://github.com/maks/e2_edit/blob/main/lib/web_midi.dart) I'm using it in at the moment.

**Comment by mortenboye** [comment]
> 

Thanks @maks, you are very welcome to submit a PR. Web support in this library would be great to have.

**Comment by mortenboye** [comment]
@maks that is great progress!
Actually there already is a repo for web support here https://github.com/InvisibleWrench/flutter_midi_command_web which you can make a PR to, but I'm happy to endorse yours as well.

**Comment by maks** [comment]
Oh sorry @mortenboye I completely missed that! Yes, I'll switch over and make a PR to your existing repo.

**Comment by maks** [comment]
One thing I should also add is I'm depending on the very new js_binding package, it then depends on the latest version of package:js which means this only works on Flutter master channel for the moment.

**Comment by mortenboye** [comment]
I think if we keep the web support on the dev branch initially, that should not be a problem.

**Comment by joranmulderij** [comment]
I really hope this package will support web. That would be great! Thanks for this package!

**Comment by maks** [comment]
Just a bit of an update as I'm returning to the this topic after quite some time as I'll need it for an upcoming project. 

While [my original PR was usable but never got merged](https://github.com/InvisibleWrench/flutter_midi_command_web/pull/2), now that Flutter via Dart has [a new JS interop API](https://dart.dev/interop/js-interop), I plan to add updated WebMidi support to flutter_midi_command_web using it.

@mortenboye would you be open to merging that once I have a PR ready?

**Comment by mortenboye** [comment]
@maks Absolutely!

**Comment by fatduckling** [comment]
Really looking forward to this, thanks very much @maks and @mortenboye <3

**Comment by mortenboye** [comment]
Fixed since v1.0.0
